src/transformer_traits.jl

```diff
@@ -360,9 +360,7 @@ function internal_type_structure(::Type{<:Pair{K,V}}, ::StructTypes.DictType) wh
     return K, V
 end
 
-function transformer(::Type{<:Pair}, ::HashVersion{4})
-    return Transformer(((a, b),) -> (a, b); hoist_type=true)
-end
+hash_trait(::Pair) = StructTypes.OrderedStruct()
```

Comment:
As discussed, this change made sense. Pairs are now treated as a OrderedStruct which is a DataType so that it could take advantage of that "path" to being hashed